src/Common/Volnitsky.h

```diff
@@ -429,6 +433,10 @@ class VolnitskyBase
 
         const auto * haystack_end = haystack + haystack_size;
 
+#ifdef __SSE4_1__
```

Comment:
Are you sure you want to invoke the fallback searcher (instead of Volnitsky) for *all* systems with SSE 4.1. That are pretty much all machines nowadays which means Volnitsky is no longer called at all.

Reply:
The Volnitsky perform good when the needle_size is very large, but not good using small needle_size. 

I have pushed another commit 'Optimize the SIMD StringSearcher if needle_size is large'. Currently, the optimized SIMD has win the Volnitsky almost all the time. The optimized SIMD StringSearcher has led at most 41.7% than Volnitsky algorithm when the needle_size is less than 21, and fallen behind only about 1% even when the needle_size is bigger than 50, which is not considered as a common case.

If the platform support X86 SSE, we could use the optimized SIMD StringSearcher.
If the platform does not support X86 SSE, then we should Use Volnitsky or the common Byte-by-Byte search (fallback). 

Test platform: ICX server
Test query: SELECT COUNT(*) FROM hits WHERE URL LIKE '%{Needle}%';

Needle_size	opt/baseline
5		141.7%
6		129.4%
8		118.5%
9		112.3%
10		107.4%
14		103.4%
20		100.2%
21		100.7%
51		99.0%

Reply:
Very interesting numbers, thank you. In the PR, you wrote that Volnitsky wins if the needle is very long (e.g. > 150 characters). Does it make sense to make the fallback choice dependent on the needle length?